File: cpp/easy/longest_continuous_increasing_subsequence.cpp
class Solution {
public:
    int findLengthOfLCIS(vector<int>& nums) {
        if(nums.size() == 0)
            return 0;
        int ret = 1;
        int cnt = 1;
        for(int i = 1; i < nums.size(); ++i)
        {
            if(nums[i] > nums[i-1]) 
                cnt++;
            else
                cnt = 1;
            ret = max(ret, cnt);
        }
        return ret;
    }
};


Conclusion:
很简单的一道题，每两个相邻的比较大小即可.

Time Complexity: O(n)
Space Complexity: O(1)

File: python/easy/longest_continuous_increasing_subsequence.py
class Solution(object):
    def findLengthOfLCIS(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 0:
            return 0
        cnt = 1
        ret = 1
        for i in xrange(1, len(nums)):
            if nums[i-1] < nums[i]:
                cnt += 1
                ret = max(ret, cnt)
            else:
                cnt = 1
        return ret
